learningresources/models.py

```diff
@@ -79,14 +82,36 @@ def has_resources(self):
 
     @transaction.atomic
     def save(self, *args, **kwargs):
-        """Handle slugs."""
+        """Handle slugs and groups"""
+        is_create = False
+        is_update = False
         if self.id is None or self.name != get_object_or_404(
                 Repository, id=self.id).name:
+            # if it is an update of the repository, need the old slug
+            if self.id is not None:
+                is_update = True
+                old_slug = self.slug
```

Comment:
Should probably change this to `old_slug = get_object_or_404(Repository, id=self.id).slug` in case caller has set slug value to something other than what it was in the database